include/deal.II/base/timer.h

```diff
@@ -732,18 +783,37 @@ Timer::get_data() const
 
 
 
+inline
+const Utilities::MPI::MinMaxAvg &
+Timer::get_total_data() const
+{
+  return mpi_total_data;
+}
+
+
+
 template <class StreamType>
 inline
 void
 Timer::print_data(StreamType &stream) const
 {
-  unsigned int my_id = dealii::Utilities::MPI::this_mpi_process(mpi_communicator);
-  if (my_id==0)
-    stream << mpi_data.max << " wall,"
-           << " max @" << mpi_data.max_index
-           << ", min=" << mpi_data.min << " @" << mpi_data.min_index
-           << ", avg=" << mpi_data.avg
-           << std::endl;
+  const Utilities::MPI::MinMaxAvg statistic = get_data();
+  stream << statistic.max << " wall,"
```

Comment:
As we usually use a ConditionalOstream object for dealing with output when using MPI it should be in the responsibility of the passed stream to decide what to do with the output. Do you agree?